mne_connectivity/tests/test_wsmi.py

```diff
@@ -0,0 +1,582 @@
+            f"tau=2 ratio should be > 2x tau=1 ratio"
+            f"tau=2 ratio / tau=1 ratio: {ratio_tau2 / ratio_tau1:.2f}"
+        )
+
+    # Both should detect some coupling
+    assert nonlinear_tau1 > 0, "tau=1 should detect some nonlinear coupling"
+    assert nonlinear_tau2 > 0, "tau=2 should detect some nonlinear coupling"
+
+
+def test_wsmi_parameter_effects():
+    """Test that different kernel and tau values produce different results."""
+    sfreq = 100.0
+    n_epochs, n_times = 2, 200
+    t = np.linspace(0, n_times / sfreq, n_times)
+
+    # Create structured data
+    data = np.zeros((n_epochs, 3, n_times))
+    for epoch in range(n_epochs):
+        data[epoch, 0, :] = np.sin(2 * np.pi * 10 * t)
+        data[epoch, 1, :] = np.sin(2 * np.pi * 10 * t + np.pi / 4)  # Phase shifted
+        data[epoch, 2, :] = np.sin(2 * np.pi * 15 * t)  # Different frequency
+
+    ch_names = ["A", "B", "C"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Test different parameter combinations
+    conn_k3_t1 = wsmi(epochs, kernel=3, tau=1)
+    conn_k4_t1 = wsmi(epochs, kernel=4, tau=1)
+    conn_k3_t2 = wsmi(epochs, kernel=3, tau=2)
+
+    # Results should be different for different parameters
+    assert not np.array_equal(conn_k3_t1.get_data(), conn_k4_t1.get_data()), (
+        "Different kernel should produce different results"
+    )
+    assert not np.array_equal(conn_k3_t1.get_data(), conn_k3_t2.get_data()), (
+        "Different tau should produce different results"
+    )
+
+    # All should have same shape and finite values
+    for conn in [conn_k3_t1, conn_k4_t1, conn_k3_t2]:
+        assert conn.get_data().shape == (n_epochs, 3)  # 3 choose 2 = 3 connections
+        assert np.all(np.isfinite(conn.get_data()))
+
+
+def test_wsmi_weighted_vs_unweighted():
+    """Test weighted parameter produces different results for wSMI vs SMI."""
+    sfreq = 100.0
+    n_epochs, n_times = 2, 200
+
+    # Create simple structured data
+    data = np.zeros((n_epochs, 3, n_times))
+    t = np.linspace(0, n_times / sfreq, n_times)
+
+    for epoch in range(n_epochs):
+        data[epoch, 0, :] = np.sin(2 * np.pi * 10 * t)
+        data[epoch, 1, :] = np.sin(2 * np.pi * 10 * t)  # Identical
+        data[epoch, 2, :] = np.sin(2 * np.pi * 12 * t)  # Different
+
+    ch_names = ["A", "B", "C"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Test wSMI vs SMI
+    conn_wsmi = wsmi(epochs, kernel=3, tau=1, weighted=True)
+    conn_smi = wsmi(epochs, kernel=3, tau=1, weighted=False)
+
+    # Should have different method names
+    assert conn_wsmi.method == "wSMI"
+    assert conn_smi.method == "SMI"
+
+    # Should produce different results due to weighting
+    assert not np.array_equal(conn_wsmi.get_data(), conn_smi.get_data()), (
+        "wSMI and SMI should produce different results"
+    )
```

Comment:
again, just checking that "results are different" is not stringent enough. At minimum it should be possible to say "one is larger than the other".  Ideally, with known inputs, it should also be possible to say *by how much* it should be larger.